INSTALL.md

```diff
@@ -1,206 +1,65 @@
+    Read the comments to guide you. For production use, it is *important*
+    that you configure the volumes appropriately to avoid data loss.
 
-The API should be exposed using WSGI server such as `waitress`, `gunicorn` or
-similar. Other configurations might be possible but I didn't pursue them.
+### Running the Application
 
-API calls are made to the relative URL `/api/`. Your HTTP server should be
-configured to proxy this URL format to the WSGI server. Some users may prefer
-to use a dedicated reverse proxy for this, to incorporate additional features
-such as load balancing and SSL.
+1. Configurations for ElasticSearch:
 
-Note that the API URL in the virtual host configuration needs to be the same as
-the one in the `config.yaml`, so that client knows how to access the backend!
+    You may need to raise the `vm.max_map_count`
+    parameter to at least `262144` in order for the
+    ElasticSearch container to function. Instructions
+    on how to do so are provided
+    [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-cli-run-prod-mode).
 
-#### Example
+2. Build or update the containers:
 
-In this example:
-
-- The booru is accessed from `http://example.com/`
-- The API is accessed from `http://example.com/api`
-- The API server listens locally on port 6666, and is proxied by nginx
-- The static files are served from `/srv/www/booru/client/public/data`
-
-**nginx configuration**:
-
-```nginx
-server {
-    listen 80;
-    server_name example.com;
-
-    location ~ ^/api$ {
-        return 302 /api/;
-    }
-    location ~ ^/api/(.*)$ {
-        if ($request_uri ~* "/api/(.*)") { # preserve PATH_INFO as-is
-            proxy_pass http://127.0.0.1:6666/$1;
-        }
-    }
-    location / {
-        root /srv/www/booru/client/public;
-        try_files $uri /index.htm;
-    }
-}
-```
-
-**`config.yaml`**:
-
-```yaml
-data_url: 'http://example.com/data/'
-data_dir: '/srv/www/booru/client/public/data'
-```
-
-To run the server using `waitress`:
+    ```console
+    user@host:szuru$ docker-compose pull
+    user@host:szuru$ docker-compose build --pull
+    ```
 
-```console
-user@host:szuru/server$ source python_modules/bin/activate
-(python_modules) user@host:szuru/server$ pip install waitress
-(python_modules) user@host:szuru/server$ waitress-serve --port 6666 szurubooru.facade:app
-```
+    This will build both the frontend and backend containers, and may take
+    some time.
 
-or `gunicorn`:
+3. Start and stop the the application
 
-```console
-user@host:szuru/server$ source python_modules/bin/activate
-(python_modules) user@host:szuru/server$ pip install gunicorn
-(python_modules) user@host:szuru/server$ gunicorn szurubooru.facade:app -b 127.0.0.1:6666
-```
+    ```console
+    # To start:
+    user@host:szuru$ docker-compose up -d
+    # To monitor (CTRL+C to exit):
+    user@host:szuru$ docker-compose logs -f
+    # To stop
+    user@host:szuru$ docker-compose down
+    ```
```

Comment:
All in all, great instructions. The comments in `docker-compose.yml.example` are also really helpful.

The only thing I'd add is the info that the frontend can be reached via `0.0.0.0:port` (the default in `docker-compose.yml` being `0.0.0.0:8080`) as people who haven't worked with Docker before might try `localhost:port` or `127.0.0.1:port` instead.

I see that the frontend is already served via nginx. I'm not that experienced with production deployment with Docker yet, but would you expose the frontend directly on `:80`/`:443` on production and adjust the configuration of the nginx running inside the container to use HTTPS, or would you still expose to `:8080` and then reverse-proxy with an nginx that's running on "bare metal" and has the HTTPS configuration?

I'm mainly curious here, but that might also be something to add (how to best go about deploying for production with HTTPS) – could also be just a link to an external guide.